docs/csharp/tutorials/nullable-reference-types.md

```diff
@@ -0,0 +1,225 @@
+
+## Create the application and enable nullable reference types
+
+Create a new console application, either in Visual Studio or from the command line using `dotnet new console`. Name the application `NullableIntroduction`. Once you've created the application, you'll need to enable C# 8 beta features. Open the `csproj` file, and add a `LangVersion` element to the `PropertyGroup` element:
+
+```xml
+<LangVersion>8.0</LangVersion>
+```
+
+Alternatively, you can use the Visual Studio project properties to enable C# 8. Right click on the project node, select **Properties**. Then, select the **build** tab, and click **Advanced...**. In the dropdown for language version, select **C# 8.0 (beta)**.
+
+You must opt into the **nullable reference types** feature, even in C# 8 projects. That's because once the feature is turned on, existing reference variable declarations become **non-nullable reference types**. While that decision will help find issues where existing code may not have proper null-checks, it may not accurately reflect your original design intent. You turn on the feature with a new pragma:
+
+```csharp
+#nullable enable
+```
+
+You can add the preceding pragma anywhere in a source file, and the nullable reference type feature is turned on from that point. The pragma also supports the `disable` argument to turn off the feature.
+
+> [!NOTE]
+> Future previews will build on the `#nullable` pragma to control the feature at a project level. In preview 1, you'll need to add this pragma to every source file.
+
+### Design the types for the application
+
+This survey application requires creating a number of classes:
+
+- A class that models the list of questions.
+- A class that models a list of people contacted for the survey.
+- A class that models the answers from a person that took the survey.
+
+These types will make use of both nullable and non-nullable reference types to express which members are required, and which members are optional. Nullable reference types communicate that design intent clearly:
+
+- The questions that are part of the survey can never be null: It makes no sense to ask a blank question.
+- The respondents can never be null. You'll want to track people you contacted, even respondents that declined to participate.
+- Any response to a question may be null. Respondents can decline to answer some, or all questions.
+
+If you've programmed in C#, you may be so accustomed to reference types that allow null values that you may have missed other opportunities to declare non-nullable instances:
+
+- The collection of questions should be non-nullable.
+- The collection of respondents should be non-nullable.
+
+As you write the code, you'll see that a non-nullable reference type as the default for references avoids common mistakes that could lead to null reference exceptions. This tutorial points out where you may have made a default decision in your C# coding.
```

Comment:
I find the second sentence confusing, particularly the hypothetical "where you may have made".